lib/wzmaplib/CMakeLists.txt

```diff
@@ -43,13 +43,6 @@ add_library(wzmaplib STATIC ${PUBLIC_HEADERS} ${PRIVATE_HEADERS} ${SRC})
 set_property(TARGET wzmaplib PROPERTY FOLDER "lib")
 set_target_properties(wzmaplib PROPERTIES VERSION "${WZMAPLIB_VERSION_STRING}")
 
-set_target_properties(wzmaplib
-	PROPERTIES
-		CXX_STANDARD 11
-		CXX_STANDARD_REQUIRED YES
-		CXX_EXTENSIONS NO
-)
-
```

Comment:
This should not be removed (as system / compiler defaults may differ), but instead should be bumped to `CXX_STANDARD 14` (same for the other instance in this file).

Reply:
But we already have `CMAKE_CXX_STANDARD 14` and friends in the root `CMakeLists.txt` file.

This acts as a default value for every other CMake target in every subdirectory (only if a directory doesn't override this, of course), so this will become `CXX_STANDARD 14` for every other target upon its creation, irrespectible of what system defaults are, actually.

Reply:
@past-due Fixed your suggestion.

Reply:
Thanks, @ManManson 👍

(One project is https://github.com/warzone2100/maptools-cli, but IIRC some people have experimented with utilizing wzmaplib for various other purposes / tools. I'm not sure if any of them are publicly available yet.)

Reply:
Thanks for the explanation! I'll definitely take a look at other places which might use `wzmaplib`.

Also, there is one more failing check from CodeQL regarding type conversions, so I'll fix this and I think this will be ready to go in.